### PR TITLE
fix(deps): pin kubernetes<36 in all images to fix kubeflow-training v1.9.3 crash

### PR DESCRIPTION
**Pull request by ChughShilpa**
Extends the [RHOAIENG-75322](https://redhat.atlassian.net/browse/RHOAIENG-75322) fix in (PR #[4092](https://github.com/opendatahub-io/notebooks/pull/4092)) to all remaining notebook images that include `kubeflow-training`. The previous PR only covered `jupyter/datascience`; this PR pins `kubernetes<36` in the remaining 7 images.

## Description
`kubeflow-training==1.9.3` crashes with `kubernetes>=36.0.0` because the kubernetes client changed its `openapi_types` format from `dict(str, str)` to `dict[str, str]`. The v1 SDK's `ApiClient.deserialize()` cannot parse the new syntax, causing an `AttributeError` when deserializing any Kubernetes object with dict-typed fields (labels, annotations, etc.).

### `pyproject.toml` — Added `kubernetes<36` override to:
- `codeserver/ubi9-python-3.12`
- `jupyter/pytorch/ubi9-python-3.12`
- `jupyter/pytorch+llmcompressor/ubi9-python-3.12`
- `jupyter/rocm/pytorch/ubi9-python-3.12`
- `jupyter/rocm/tensorflow/ubi9-python-3.12`
- `jupyter/tensorflow/ubi9-python-3.12`
- `jupyter/trustyai/ubi9-python-3.12`

### Lock files & manifests
- Regenerated `requirements.*.txt` and `pylock.*.toml` for all affected images
- Updated imagestream annotations to reflect resolved versions

## How Has This Been Tested?
- Removed the `-datascience-` skip condition from `test_kubeflow_training` in `libraries_testunits.py` so the test covers all images that run this test file

## Verification
No cascade downgrades across any image — `kfp`, `boto3`, and `requests` remain at the same versions as `main`:

| Image | kubernetes | kfp | boto3 |
|---|---|---|---|
| codeserver | 35.0.0 | 2.17.0 | 1.43.46 |
| pytorch | 35.0.0 | 2.17.0 | 1.43.46 |
| pytorch+llmcompressor | 35.0.0 | 2.17.0 | 1.43.46 |
| rocm/pytorch | 35.0.0 | 2.17.0 | 1.43.46 |
| rocm/tensorflow | 35.0.0 | 2.17.0 | 1.43.46 |
| tensorflow | 35.0.0 | 2.17.0 | 1.43.46 |
| trustyai | 35.0.0 | 2.17.0 | 1.43.46 |

Self checklist (all need to be checked):
- [ ] Ensure that you have run `make test` (`gmake` on macOS) before asking for review
- [ ] Changes to everything except `Dockerfile.konflux` files should be done in `odh/notebooks` and automatically synced to `rhds/notebooks`. For Konflux-specific changes, modify `Dockerfile.konflux` files directly in `rhds/notebooks` as these require special attention in the downstream repository and flow to the upcoming RHOAI release.

## Merge criteria:
- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

[RHOAIENG-75322]: https://redhat.atlassian.net/browse/RHOAIENG-75322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated Kubernetes support across container environments to use version 35.0.0, preventing compatibility issues with Kubeflow Training.
  * Applied the compatibility safeguard consistently across CPU, CUDA, and ROCm configurations.

* **Tests**
  * Expanded Kubeflow Training validation to run across all applicable image types.

### CODE REVIEW

**Review by sutaakar — APPROVED** [review]

File: codeserver/ubi9-python-3.12/pyproject.toml
Comment (jiridanek) [inline codeserver/ubi9-python-3.12/pyproject.toml:1]: @coderabbitai Checked out locally on `RHOAIENG-75322-allImages` (`301a955`). Review of [PR #4126](https://github.com/opendatahub-io/notebooks/pull/4126):

## Verdict

**Approve the fix, but block merge until rebase.** The dependency change is correct and complete for workbench images that ship `kubeflow-training`. Merge is currently impossible (`mergeable: CONFLICTING`).

## What looks good

- Extends [#4092](https://github.com/opendatahub-io/notebooks/pull/4092) consistently: `kubernetes<36` override + lock regeneration for the remaining 7 images that include `kubeflow-training`.
- Coverage of that package is complete across workbench `pyproject.toml`s (datascience already pinned; this PR finishes the set). Runtimes still resolve `kubernetes==36.0.2` but do **not** ship `kubeflow-training` — fine.
- Removing the `-datascience-` skip in `test_kubeflow_training` is the right follow-up so non-minimal Jupyter images exercise the deserialize path.
- Author’s cascade check looks credible: only `kubernetes` moves `36.0.2 → 35.0.0`; `kfp` / `boto3` stay put. Most affected images didn’t need the datascience-only `requests==2.33.0` pin.
- Jupyter ODH builds that completed (pytorch, tensorflow, trustyai, datascience, …) succeeded with `kubernetes==35.0.0` installed.

## Findings

### 1. Blocking — rebase required (medium)

Conflicts with `main` in **18** `pylock.*.toml` files — essentially competing `--exclude-newer=` timestamps (this PR `…T14:10:31Z` vs main `…T11:35:02Z` from later lock refreshes). Needs rebase + `make refresh-pipfilelock-files` (or equivalent) before merge.

### 2. Codeserver CI red — likely infra, not this change (low / note)

`codeserver` amd64/arm64 and Konflux codeserver failed. Failure signature is a CentOS Stream mirror 404 (`modules.yaml.xz` on aarch64), not a dep/resolve error. Retry after rebase; don’t treat as a product regression from the pin.

### 3. Test gap for codeserver (low)

`libraries_testunits.py` only runs via `libraries_test.py` → `jupyterlab_datascience_image` (non-minimal Jupyter). Codeserver gets the pin but **not** `test_kubeflow_training`. Acceptable for a stopgap; worth a follow-up if you care about codeserver KFTO parity.

### 4. PR body inaccuracy (nit)

Claims “Updated imagestream annotations” — no imagestream/manifest files in the diff. Also checklist boxes for `make test` / sync note are unchecked.

### 5. Lockfile noise (nit)

Timestamp-only churn on `jupyter/minimal` and `runtimes/*/pylock.*` with no version changes. Harmless but widens the conflict surface on every refresh.

## CI snapshot

| Area | Status |
|------|--------|
| Static analysis / CodeQL / Semgrep | Green |
| Most ODH image builds + Konflux Jupyter/runtimes | Green |
| codeserver (GHA + Konflux) | Red (infra) |
| RHOAI subscription / AIPCC builds | Skipped (fork) |
| Mergeability | **Dirty / conflicting** |
| Existing review | `lgtm` from @sutaakar |

## Recommendation

Ask the author to **rebase on `main`, regenerate locks, push**, then re-run CI (especially codeserver). After that, this is a straightforward approve/merge for the RHOAIENG-75322 stopgap (remove in 3.6 EA1 per the TODO / linked follow-up).

Want me to post these as inline PR comments (with `@coderabbitai`) or leave an approve/comment review on GitHub?

File: codeserver/ubi9-python-3.12/uv.lock.d/pylock.cpu.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
...
+#    uv pip compile pyproject.toml --output-file uv.lock.d/pylock.cpu.toml --format pylock.toml --generate-hashes --emit-index-url --python-version=3.12 --universal --no-annotate --no-emit-package odh-notebooks-meta-db-connectors-deps --no-emit-package odh-notebooks-meta-jupyterlab-datascience-deps --no-emit-package odh-notebooks-meta-jupyterlab-deps --no-emit-package odh-notebooks-meta-llmcompressor-deps --no-emit-package odh-notebooks-meta-runtime-elyra-deps --no-emit-package odh-notebooks-meta-runtime-datascience-deps --no-emit-package odh-notebooks-meta-runtime-kale-deps --no-emit-package odh-notebooks-meta-workbench-datascience-deps --constraints ../../dependencies/cve-constraints.txt --exclude-newer=2026-07-21T14:10:31Z --default-index=https://packages.redhat.com/api/pypi/public-rhai/rhoai/3.5/cpu-ubi9/simple/?format=json
```
Comment (jiridanek) [inline codeserver/ubi9-python-3.12/uv.lock.d/pylock.cpu.toml:2]: **Blocking:** Rebase onto `main` and regenerate lockfiles before merge.

This PR is currently `CONFLICTING` with `main`. The conflicts are concentrated in `pylock.*.toml` headers like this one (competing `--exclude-newer=` timestamps from later lock refreshes on `main`). After rebase, re-run lock regeneration (e.g. `make refresh-pipfilelock-files`) so resolved versions/hashes are revalidated in CI.

This is the only blocking item for merge.

@coderabbitai — can you confirm the merge conflicts are limited to these pylock header timestamp clashes after comparing against current `main`?

**Review by jiridanek — COMMENTED** [review]
## Review

Compatibility pin looks correct as a temporary override for the `kubeflow-training==1.9.3` / kubernetes-client 36.x `openapi_types` crash (not a CVE/CWE fix). Still treat the dep/lock churn as supply-chain-sensitive: rebase, regenerate locks, and revalidate in CI.

### Blocking
See the inline note on a representative `pylock.*.toml` header — rebase onto `main` and regenerate locks before merge (PR is currently conflicting).

### Non-blocking

1. **codeserver CI (GHA + Konflux)** — Retry after rebase. Current failure evidence points at CentOS Stream mirror infrastructure (`modules.yaml.xz` 404), not the `kubernetes<36` resolution.
2. **codeserver deserialize-test coverage** — Follow-up only. `test_kubeflow_training` runs via Jupyter `libraries_test.py`, so codeserver is pinned but not covered by that deserialize assertion.
3. **PR body / checklist** — Description claims imagestream annotation updates, but none appear in the diff; `make test` / sync checklist boxes are unchecked.
4. **Timestamp-only lockfile churn** — Several `jupyter/minimal` and `runtimes/*/pylock.*` files only refresh `--exclude-newer=` timestamps. Harmless, but it widens rebase conflict surface.

@coderabbitai — please keep framing this as an API-format compatibility stopgap for kubeflow-training 1.9.3, not a security remediation, when summarizing.

File: codeserver/ubi9-python-3.12/uv.lock.d/pylock.cpu.toml (hunk 1 above)
Comment (jiridanek) [reply codeserver/ubi9-python-3.12/uv.lock.d/pylock.cpu.toml:2]: @coderabbitai fixed conflicts, recheck

**Review by jiridanek — APPROVED** [review]

**Comment by jiridanek** [comment]
## CI status — [PR #4126](https://github.com/opendatahub-io/notebooks/pull/4126)

**Branch / SHA:** `RHOAIENG-75322-allImages` @ `83c75ad`  
**Mergeable:** yes (git), but tide blocked by Konflux. Labels: `approved` + `lgtm`.

### Konflux (path-filtered — 7 pipelines for this SHA)

| Pipeline | Archive | GitHub check |
|----------|---------|--------------|
| codeserver-datascience-cpu | Completed | success |
| jupyter-pytorch-cuda | Completed | success |
| jupyter-pytorch-llmcompressor-cuda | Completed | success |
| jupyter-tensorflow-cuda | Completed | success |
| jupyter-tensorflow-rocm | Completed | success |
| jupyter-pytorch-rocm | **Completed** (19:39Z) | still **in_progress** (stale PAC report) |
| jupyter-trustyai-cpu | **Failed** | **failure** |

### Trustyai root cause (infra, not the pin)

Failed task: `build-images` / `step-build` on **linux-d160-m2xlarge/arm64**:

```text
Error allocating host: failed to provision host
Platform: linux-d160-m2xlarge/arm64
```

Image build itself did not fail; the multi-arch build host never came up. Unrelated to `kubernetes<36`.

### Recommended next step

Retrigger trustyai (and optionally pytorch-rocm to clear the stale check), e.g.:

```text
/kfbuild odh-workbench-jupyter-trustyai-cpu-py312-ubi9
```

or `/kfbuild all` if you want a full refresh. After trustyai goes green, tide should be able to merge (codeserver already succeeded; its tide message looks stale).
